klippy/extras/input_shaper.py

```diff
@@ -123,25 +123,41 @@ def parse_shaper(shaper_type_str):
 
         shaper_type = gcmd.get('SHAPER_TYPE', None, parser=parse_shaper)
         if shaper_type is None:
-            shaper_type_x = gcmd.get('SHAPER_TYPE_X', self.shaper_type_x,
+            shaper_type_x = gcmd.get('SHAPER_TYPE_X', None,
                                      parser=parse_shaper)
-            shaper_type_y = gcmd.get('SHAPER_TYPE_Y', self.shaper_type_y,
+            shaper_type_y = gcmd.get('SHAPER_TYPE_Y', None,
                                      parser=parse_shaper)
         else:
             shaper_type_x = shaper_type_y = shaper_type
 
-        self._set_input_shaper(shaper_type_x, shaper_type_y,
-                               shaper_freq_x, shaper_freq_y,
-                               damping_ratio_x, damping_ratio_y)
+        if damping_ratio_x is not None:
```

Comment:
I'm afraid that writing the new values directly into the member variables will break new_delay / old_delay  calculations and the subsequent call to `toolhead.note_step_generation_scan_time` in `_set_input_shaper` method. This may manifest itself as the errors in the stepcompress code upon changes of the shaper frequency or the shaper type. So, this isn't really the right way to do it.

Reply:
oh ok. so if i reverse it to:

```python
if damping_ratio_x is None:
  damping_ratio_x = self.damping_ratio_x
```
and check before for the output, it would fit?

Reply:
Yeah, that should work.